packages/react/src/primitives/PhoneNumberField/PhoneNumberField.tsx

```diff
@@ -28,10 +29,40 @@ const PhoneNumberFieldPrimitive: Primitive<PhoneNumberFieldProps, 'input'> = (
     onInput,
     size,
     variation,
+    defaultValue,
+    onChange,
+    value,
     ...rest
   },
   ref
 ) => {
+  // state for the current dial code and default to the defaultDialCode
+  const [dialCode, setDialCode] = React.useState<string>(
+    (defaultDialCode as string) ?? ''
+  );
+  // state for the formatted phone number and default to the defaultValue
+  const [formattedPhoneNumber, setFormattedPhoneNumber] =
+    React.useState<string>(
+      formatPhoneNumber(dialCode, (value as string) ?? defaultValue ?? '')
+    );
+  const onNumberChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const phoneNumber = event.target.value;
+    // format the phone number with the current dial code
+    const formatted = formatPhoneNumber(dialCode, phoneNumber);
+    setFormattedPhoneNumber(formatted);
+    onChange?.(event);
+  };
+  const onDialCodeChangeWrapper = (
+    event: React.ChangeEvent<HTMLSelectElement>
+  ) => {
+    const code = event.target.value;
+    // Call the onDialCodeChange prop if it exists
```

Comment:
Huh?

Reply:
This is used to detect when dialCode changes to apply the formatting with the new Code, you can see it in action in the video I attached above.

Reply:
FYI code comments are usually placed directly above what the comment is referencing, suggest applying that here